source/common/upstream/health_checker_impl.cc

```diff
@@ -169,16 +171,52 @@ HttpHealthCheckerImpl::HttpStatusChecker::HttpStatusChecker(
           fmt::format("Invalid http status range: expecting end <= 600, but found end={}", end));
     }
 
-    ranges_.emplace_back(std::make_pair(static_cast<uint64_t>(start), static_cast<uint64_t>(end)));
+    expected_ranges_.emplace_back(
+        std::make_pair(static_cast<uint64_t>(start), static_cast<uint64_t>(end)));
   }
 
-  if (ranges_.empty()) {
-    ranges_.emplace_back(std::make_pair(default_expected_status, default_expected_status + 1));
+  if (expected_ranges_.empty()) {
+    expected_ranges_.emplace_back(
+        std::make_pair(default_expected_status, default_expected_status + 1));
   }
+
+  for (const auto& status_range : retriable_statuses) {
+    const auto start = status_range.start();
+    const auto end = status_range.end();
+
+    if (start >= end) {
+      throw EnvoyException(fmt::format("Invalid http retriable status range: expecting start < "
+                                       "end, but found start={} and end={}",
+                                       start, end));
+    }
+
+    if (start < 100) {
+      throw EnvoyException(fmt::format(
+          "Invalid http retriable status range: expecting start >= 100, but found start={}",
+          start));
+    }
+
+    if (end > 600) {
+      throw EnvoyException(fmt::format(
+          "Invalid http retriable status range: expecting end <= 600, but found end={}", end));
+    }
+
+    retriable_ranges_.emplace_back(
```

Comment:
Can we wrapper this redundant range-checking codes to a validate func or some thing?